test/common/network/filter_manager_impl_test.cc

```diff
@@ -88,15 +88,15 @@ TEST_F(NetworkFilterManagerTest, All) {
 
   write_buffer_.add("foo");
   write_end_stream_ = false;
-  EXPECT_CALL(*write_filter, onWrite(BufferStringEqual("foo"), false))
+  EXPECT_CALL(*filter, onWrite(BufferStringEqual("foo"), false))
```

Comment:
Can you add a test for reversed order?

Reply:
All tests here are for reversed order because I set 
bool reverseWriteFilterOrder() const override { return true; }
in network/mocks.

Reply:
I am not following what you were suggesting. Did you mean a test with the reverseWriteFilterOrder returning false, e.g., when the order is NOT reversed? Like what I did for the http filters here, e.g., https://github.com/envoyproxy/envoy/blob/bacaa4cdcbf94a8b110b783ae5e72cd2947a8432/test/common/http/conn_manager_impl_test.cc#L499

Reply:
Yes I'm suggesting that you test when someone changes the option to false. It would be good to add such a test.

Reply:
Done.